packages/iris-grid/src/IrisGrid.tsx

```diff
@@ -3171,42 +3176,88 @@ export class IrisGrid extends Component<IrisGridProps, IrisGridState> {
 
   async seekRow(inputString: string, isBackwards = false): Promise<void> {
     const {
-      gotoValueSelectedColumn: selectedColumn,
+      gotoValueSelectedColumnName: selectedColumnName,
       gotoValueSelectedFilter,
-      lastSeekedRow,
     } = this.state;
     const { model } = this.props;
-    if (isIrisGridProxyModel(model)) {
-      const { table } = model;
-      if (table !== undefined && !TableUtils.isTreeTable(table)) {
-        if (selectedColumn.type === 'java.lang.String') {
-          // is string column,
-          const isContains = gotoValueSelectedFilter === 1;
-          const rowIndex = await table.seekRow(
-            isBackwards ? lastSeekedRow - 1 : lastSeekedRow + 1,
-            selectedColumn,
-            'String',
+    const columnIndex = model.getColumnIndexByName(selectedColumnName);
+    if (columnIndex === undefined) {
+      return;
+    }
+
+    const selectedColumn = model.columns[columnIndex];
+
+    if (
+      !isIrisGridProxyModel(model) ||
+      model.table === undefined ||
+      TableUtils.isTreeTable(model.table)
+    ) {
+      return;
+    }
+
+    let searchFromRow;
+
+    if (this.grid) {
+      ({ selectionEndRow: searchFromRow } = this.grid.state);
+    }
+
+    if (searchFromRow == null) {
+      searchFromRow = 0;
+    }
+
+    const { table } = model;
+    const isContains = gotoValueSelectedFilter === FilterType.contains;
+    const isEquals =
+      gotoValueSelectedFilter === FilterType.eq ||
+      gotoValueSelectedFilter === FilterType.eqIgnoreCase;
+
+    try {
+      if (selectedColumn.type === 'java.lang.String') {
+        // is string column,
+        const rowIndex = await table.seekRow(
+          isBackwards ? searchFromRow - 1 : searchFromRow + 1,
+          selectedColumn,
+          'String',
+          inputString,
+          isEquals,
+          isContains,
+          isBackwards ?? false
+        );
+        this.grid?.setFocusRow(rowIndex);
+      } else if (inputString !== '') {
+        const columnDataType = TableUtils.getSeekRowColumnType(selectedColumn);
+        let rowIndex;
+        if (columnDataType === 'DateTime') {
+          const { formatter } = model;
+          const [startDate] = DateUtils.parseDateRange(
             inputString,
-            false,
-            isContains,
+            formatter.timeZone
+          );
+          rowIndex = await table.seekRow(
```

Comment:
This still needs to be abstracted through `IrisGridModel`, we should not be calling anything on the `Table` object directly.